### PR TITLE
Fix 5856 for Export-DbaRegServer

### LINKED ISSUE #5856: FilePath not honored in Export-DbaRegServer
it keeps being a directory

### PR DESCRIPTION
**Pull request by lancasteradam**
- Fixes #5856 by ensuring -FilePath is implemented
- Added -Group and -ExcludeGroup params
- Changed the command to allow for either .xml or .regsrvr file extensions
- Updated documentation for the new params
- Added more integration tests
- Fixed the implementation of -ExcludeGroup in Get-DbaRegServerGroup
- Fixed the Disconnect call in Add-DbaRegServerGroup

## Type of Change

 - [x] Bug fix (non-breaking change, fixes #5856  )
 - [x] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 

### Purpose
 
See comments above.

### Commands to test

See the pester test files.

### Learning

Helpful resource:
https://4sysops.com/archives/validating-file-and-folder-paths-in-powershell-parameters/

### CODE REVIEW

**Comment by lancasteradam** [comment]
The tests for the code involved in this PR have completed successfully. The errors seem to be from Get-DbaDbSharePoint.

**Comment by potatoqualitee** [comment]
Thanks so much! Re-running tests and will examine shortly.

**Comment by potatoqualitee** [comment]
this looks fantastic to me! thank you so much. the additional tests are much appreciated. @ClaudioESSilva, any chance you can take a look and approve as well?

**Comment by potatoqualitee** [comment]
actually, i do see one issue. will comment now 👍

File: functions/Export-DbaRegServer.ps1 [hunk 1]
```diff
@@ -73,47 +82,96 @@ function Export-DbaRegServer {
...
-        $timeNow = (Get-Date -uformat "%m%d%Y%H%M%S")
+        $timeNow = (Get-Date -UFormat "%m%d%Y%H%M%S")
+
+        # ValidateScript in the above param block relies on the order of the params specified by the user,
+        # so the creation of the file path and $Overwrite are evaluated here
+        if ($PSBoundParameters.ContainsKey("FilePath")) {
+            if (-not (Test-Path $FilePath) ) {
+                New-Item -Path $FilePath.DirectoryName -ItemType "directory" -Force | Out-Null # make sure the parent dir exists
+            } elseif (-not $Overwrite.IsPresent) {
+                throw [System.Management.Automation.RuntimeException] "Use the -Overwrite parameter if the file $FilePath should be overwritten."
```
Comment (potatoqualitee) [inline functions/Export-DbaRegServer.ps1:109]: We use an internal command, `Stop-Function`, in place of throws. `Stop-Function` is user friendly in appearance and also does additional logging behind the scenes.

```suggestion
                Stop-Function -Message "Use the -Overwrite parameter if the file $FilePath should be overwritten."
                return
```

File: functions/Export-DbaRegServer.ps1 [hunk 2]
```diff
@@ -73,47 +82,96 @@ function Export-DbaRegServer {
         [object[]]$InputObject,
         [string]$Path = (Get-DbatoolsConfigValue -FullName 'Path.DbatoolsExport'),
         [Alias("OutFile", "FileName")]
-        [string]$FilePath,
+        [ValidateScript( {
+                if ($_.FullName -notmatch "\.xml$|\.regsrvr$") {
+                    throw "The FilePath specified must end with either .xml or .regsrvr"
```
Comment (potatoqualitee) [inline functions/Export-DbaRegServer.ps1:87]: Please perform validation within the `begin` block and use `Stop-Function` in place of `throw` 🙇

**Review by potatoqualitee — CHANGES_REQUESTED** [review]

**Comment by lancasteradam** [comment]
ok, sounds good. I'll make those changes.

**Comment by potatoqualitee** [comment]
Should be done today 💯

**Comment by potatoqualitee** [comment]
let's do this 😅 !

**Pull request merged**